query.py

```diff
@@ -93,12 +93,19 @@ def _list_gen(self, list_name, key, http, url):
         while list_node.has_current():
             cur = list_node.current()
             # Yield current item.
-            yield None if cur is None else cur[key]
+            if cur is None:
+                yield None
+            elif not node_attributes:
+                yield cur[key]
+            else:
+                yield (cur[key],) + \
```

Comment:
why do you choose to return different types from this function? 
e.g. sometimes a scalar, sometimes a list (sometimes None)